web/src/components/exam/assignment/AssignmentCard.tsx

```diff
@@ -49,16 +54,20 @@ export const AssignmentCard = ({ language, assignment, exam, refreshData }: Assi
   const isPuhvi = isPuhviAssignment(assignment, exam)
   const isLd = isLdAssignment(assignment, exam)
 
+  const returnLocation = isFavoritePage
+    ? getFavoritesPagePath(exam, location.search)
+    : getContentListPath(exam, ContentType.koetehtavat, location.search)
```

Comment:
Toimisko tässä:

```suggestion
  const returnLocation = `${location.pathname}${location.search}${location.hash}`
```

Sit ei tarviis `isFavoritePage`-propsua. Ja jos tulee lisää paikkoja, joissa on AssignmentCardeja, niin palattais tehtävästä aina sinne ilman lisämuutoksia.

(Ja ei kai käytetä url-hasheja eli fragmenteja ainakaan vielä, mut kaipa sen voisi silti säästää tässä).